### ISSUE TITLE
Virtualbox DHCP Issue :: Magically NOT working

### ISSUE THREAD

**Issue by blaggacao** [docker/machine#1611]
Actually, I don't think it's a docker-machine issue, but rather one of vitrualbox. But as beeing part of the tooling, I might report here FYI, as well.

In my case after an update from Version 4.3.28 to 4.3.30, this broke the DHCP servers. II still couldn't figure out what's happening. Symptoms are, that docker machine commands show the following message:

`Unexpected error getting machine url: exit status 255`
or
`error getting URL for host MACHINENAME: exit status 255`

`VBoxManage list hostonlyifs` yields (in my case):
```
Name:            VirtualBox Host-Only Ethernet Adapter
GUID:            cb5410d0-f723-498c-a93b-becf9e18e4fa
DHCP:            Disabled
IPAddress:       192.168.57.1
NetworkMask:     255.255.255.0
IPV6Address:     fe80:0000:0000:0000:c058:474d:8c15:88f3
IPV6NetworkMaskPrefixLength: 64
HardwareAddress: 08:00:27:00:3c:33
MediumType:      Ethernet
Status:          Up
VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter
```

***Note, that*** DHCP is shown as disabled. This result is also yielded by the Windows Repair tool run on this adapter, but it cannot fix it automatically (probably, confirm, if this is the case with you, having a similar problem, as well).

On the other hand:
`VBoxManage.exe list dhcpservers` yields a bunch of legacy:
```
NetworkName:    HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter #3
IP:             192.168.99.2
NetworkMask:    255.255.255.0
lowerIPAddress: 192.168.99.100
upperIPAddress: 192.168.99.254
Enabled:        Yes

NetworkName:    HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter #4
IP:             192.168.99.1
NetworkMask:    255.255.255.0
lowerIPAddress: 192.168.99.100
upperIPAddress: 192.168.99.254
Enabled:        Yes

NetworkName:    HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter
IP:             192.168.57.100
NetworkMask:    255.255.255.0
lowerIPAddress: 192.168.57.101
upperIPAddress: 192.168.57.254
Enabled:        Yes
```
Note, that it pretends that the DHCP correpsonding to the Adapter indeed shows as "enablend".

Seems, like this issue has been discussed here, but has been reported (on 19 Feb 2011, 03:06) as fixed in 4.0.4: https://forums.virtualbox.org/viewtopic.php?f=6&t=37402

**Comment by tehmaspc** [comment]
Wondering if this is a duplicate of https://github.com/docker/machine/issues/1591 ??? Can you confirm?

**Comment by blaggacao** [comment]
I managed to figure out, that you also have the relevant line of this issue in your debug transcript: 
`error getting URL for host dev: exit status 255`, but I'm not able to connect the loose ends, in order to be able to confirm identity. Can you help with a hint?

**Comment by blaggacao** [comment]
Strange thing for the record, when removing DHCP, manually, with `"/cygdrive/c/Program Files/Oracle/VirtualBox/VBoxManage.exe" dhcpserver remove --ifname "VirtualBox Host-Only Ethernet Adapter #2"`, only those, that have an existing corresponding adapter can be removed, the others show:
```
VBoxManage.exe: error: The host network interface with the given name could not be found
VBoxManage.exe: error: Details: code E_INVALIDARG (0x80070057), component Host, interface IHost, callee IUnknown
VBoxManage.exe: error: Context: "FindHostNetworkInterfaceByName(Bstr(pIfName).mutableRaw(), hif.asOutParam())" at line 346 of file VBoxManageDHCPServer.cpp
VBoxManage.exe: error: Could not find interface 'VirtualBox Host-Only Ethernet Adapter #4'
```

However, their DHCPs still exists, will try now to recreate and adaptors, then remove dhcps and finally clean up the adapters again.

**Comment by blaggacao** [comment]
```
$ dm start b2d
Starting VM...
Too many retries.  Last error: Maximum number of retries (60) exceeded
```

**Comment by blaggacao** [comment]
Hahaha, that was a ssh permission issue on the .ssh/config I can't believe it. Although, I updated to VirtualBox 5.0 in parallel. But I-m rather sure of the ssh issue, as the VirtualBox documentation doesn't say anything about a posibility of getting the ip via VBoxManage, so I assume, that it is retireved via ssh by docker-machine. This would then close the loose ends and raise the possibility, that there is core issue identity with #1591

**Comment by blaggacao** [comment]
Maybe, we need better error messages, to be kind with human beeings.

**Comment by tehmaspc** [comment]
@blaggacao "so I assume, that it is retireved via ssh by docker-machine" - yup; it is.

cool.

**Comment by qguv** [comment]
Googlers: I also had this issue. I never did discover the cause, but I solved it by removing the VM (and its data) manually from Virtualbox and letting Docker recreate it.

**Comment by akochlef** [comment]
VBoxManage dhcpserver remove --ifname "VirtualBox Host-Only Ethernet Adapter #3"
